extensions/amp-document-recommendations/0.1/amp-document-recommendations.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @const */
+const MAX_ARTICLES = 2;
+
+/** @const */
+const SEPARATPOR_RECOS = 3;
+
+import {assertConfig} from './config';
+import {user} from '../../../src/log';
+import {tryParseJson} from '../../../src/json';
+import {CSS} from '../../../build/amp-document-recommendations-0.1.css';
+import {isJsonScriptTag} from '../../../src/dom';
+import {Services} from '../../../src/services';
+import {MultidocManager} from '../../../src/runtime';
+import {setStyle} from '../../../src/style';
+
+export class AmpDocumentRecommendations extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    this.element.classList.add('i-amp-document-recommendations');
+
+    // TODO(emarchiori): Consider using a service instead of singleton.
+    if (this.win.CONTENT_DISCOVERY) {
```

Comment:
I assume this is simply a flag so that the component is only rendered once? If so, just having it be a variable outside of class (e.g. at extension level) would be better for now until this becomes a window level service.

Reply:
Correct. Done.